kobo/apps/mass_emails/user_queries.py

```diff
@@ -1,13 +1,57 @@
+from datetime import timedelta
 from math import inf
 
-from django.db.models import QuerySet
+from django.db.models import Q, QuerySet
+from django.utils.timezone import now
 
 from kobo.apps.kobo_auth.shortcuts import User
 from kobo.apps.organizations.models import Organization
+from kobo.apps.openrosa.apps.logger.models import Instance
 from kobo.apps.stripe.utils import get_organization_plan_limits
+from kpi.models import Asset
 from kpi.utils.usage_calculator import get_storage_usage_by_user_id
 
 
+def get_inactive_users(days: int = 365) -> QuerySet:
+    """
+    Retrieve users who have been inactive for a specified number of days.
+
+    A user is considered inactive if:
+    - They have not logged in within the given period (or never logged in).
+    - They have not modified or created an XForm within the given period.
+    - They have not modified or have a submission within the given period.
+
+    :param days: int: Number of days to determine inactivity (default: 365 days)
+
+    :return: A queryset of inactive users
+    """
+    # Calculate the user inactivity threshold date
+    days = days or 365
+    inactivity_threshold = now() - timedelta(days=days)
+
+    # Identify users who have not logged in within the given period
+    inactive_users = User.objects.filter(
+        Q(last_login__lt=inactivity_threshold) | Q(last_login__isnull=True)
```

Comment:
This will catch new users who haven't logged in yet, which we don't want. Might need to check creation date as well.